python/ray/rllib/agents/trainer.py

```diff
@@ -427,6 +472,20 @@ def _init(self, config, env_creator):
 
         raise NotImplementedError
 
+    @DeveloperAPI
+    def _evaluate(self):
+        """Evaluates current policy with [possibly] altered
+        Trainer configuration settings.
+        Note that this default implementation does not
+        disables exploration for evaluation runs."""
+        logger.info("Evaluating current policy for {} episodes".format(
+            self.config["evaluation_num_episodes"]))
+        self.evaluation_ev.restore(self.local_evaluator.save())
+        for _ in range(self.config["evaluation_num_episodes"]):
+            self.evaluation_ev.sample()
+        metrics = collect_metrics(self.evaluation_ev)
+        return {"evaluation": metrics}
```

Comment:
This isn't implemented right? I would raise NotImplementedError("Please set evaluation_config") or similar to avoid confusing users.

Reply:
@ericl , could you please elaborate on this comment?

Reply:
I made a patch here. Basically, I think it is misleading to allow evaluation without specifying a config, since it will be returning the same metrics as normal policy evaluation.